docsrc/content/type-reader.fsx

```diff
@@ -89,3 +89,40 @@ and
                                         let! name = resolve t
                                         let! value = resolve d
                                         return (name,value) }
+
+
+(**
+Another usage of the Reader monad is an alternative to dependency injection or currying
+in order to pass around dependencies. The below code comes from [F# Online - Josef Starýchfojtů - FSharpPlus - Advanced FP concepts in F#](https://www.youtube.com/watch?v=pxJCHJgG8ws). You can find the presenter on github as [@starychfojtu](https://github.com/starychfojtu).
+
+Why would you want to do this style?
+
+- When you want to pass around a single environment instead of using dependency injection.
+
+Why wouldn't you want to use this style?
+
+- The downside of this style is that it supposes that your environment is relatively immutable. If you have different lifetimes for different implementation classes dependency injection frameworks can be easier to use.
```

Comment:
Also note above the consideration of the 'State' related type.

Reply:
How do you mean? That you could instead weave lifetime scoped entities in 'State'?

Reply:
You said: `it supposes that your environment is relatively immutable.` State type allows you to set and get, effectively mutating.

...and above says:

 - [State](type-state.html): Similar, but it allows you to modify the environment.

Reply:
Yes, that's true. Though the lifetime issues probably applies to State as well. I guess the solution when you have a reader environment would be to have a proxy so that you control the lifetime separate from the normal business logic flow?